src/main/java/io/github/thebusybiscuit/slimefun4/api/researches/Research.java

```diff
@@ -249,7 +249,9 @@ public boolean canUnlock(@Nonnull Player p) {
         }
 
         boolean creativeResearch = p.getGameMode() == GameMode.CREATIVE && Slimefun.getRegistry().isFreeCreativeResearchingEnabled();
-        return creativeResearch || p.getLevel() >= cost;
+        boolean economyResearch = Slimefun.getIntegrations().isVaultInstalled() && Slimefun.getIntegrations().getVaultIntegration().hasBalanceForResearch(p, cost);
+        boolean expResearch = !Slimefun.getRegistry().isVaultEconomyEnabled() && p.getLevel() >= cost;
+        return creativeResearch || economyResearch || expResearch;
```

Comment:
This is a method with high traffic, so think about if we really need to calculate all of this everytime...
If you split this into smaller methods then the `||` operator would ensure that the sub-methods are not called unnecessarily.